### ISSUE TITLE
Failed loading image/segmentation: slice indices must be integers or None or have an __index__ method in Python 3.5

### ISSUE THREAD

**Issue by waleedgondal** [hangzhaomit/semantic-segmentation-pytorch#2]
TLDR; For python 3.5+ users, In [_scale_and_crop function](https://github.com/hangzhaomit/semantic-segmentation-pytorch/blob/master/dataset.py), change the computation of x1 and y1 coordinates to 
`x1 = (w_s - cropSize) // 2 ;
 y1 = (h_s - cropSize) // 2` 
as single slash division (/) in py3.5 returns a float instead of integer in py2.7 therefore double slashes are required.

@hangzhaomit First of all thanks for this repository. The above error invokes the exception in [dataset reader](https://github.com/hangzhaomit/semantic-segmentation-pytorch/blob/master/dataset.py) and makes the dummy segmentation masks of -1 value. Now if by any chance, none of the segmentation masks in a batch get read (for e.g. due to above stated py3.5 issue), we end up getting the following error 
```
    File "/lustre/home/wgondal/semantic-segmentation-pytorch/utils.py", line 99, in accuracy
    acc = 1.0 * torch.sum(valid * (preds == segs)) / torch.sum(valid)
ZeroDivisionError: float division by zero
```
The reason is the following line in [accuracy ftn in utils](https://github.com/hangzhaomit/semantic-segmentation-pytorch/blob/master/utils.py)
`valid = (segs >= 0)`

It rightly checks for the segmentation masks value. However it would be great if you could provide a check if all the segmentation masks in a batch are dummy. OR In my opinion, the exception in datasets is misleading as it doesn't lead to figuring out the right error.

**Comment by hangzhaomit** [comment]
@waleedgondal  Thanks very much for pointing out this issue! Just fixed.
